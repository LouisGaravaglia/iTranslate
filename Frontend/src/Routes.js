import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Home from './Home';
import Browse from "./Browse";

const Routes = () => {
    return (
      <BrowserRouter>
        <div className="main">
          <Switch>
            <Route exact path="/"><Home /></Route>
            <Route exact path="/browse"><Browse /></Route>

          </Switch>
        </div>
      </BrowserRouter>
    );
  };

export default Routes;