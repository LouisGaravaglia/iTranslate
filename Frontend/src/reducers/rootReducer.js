import {combineReducers} from "redux";
import translationReducer from "./translationReducer";
import errorsReducer from "./errorsReducer";
import lyricsReducer from "./lyricsReducer";
import resultsReducer from "./resultsReducer";

const rootReducer = combineReducers({results: resultsReducer, translation: translationReducer, lyrics: lyricsReducer, errors: errorsReducer});

export default rootReducer;