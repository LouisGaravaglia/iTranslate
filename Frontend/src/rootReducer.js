// const SEED_ARTISTS = [{name: "J Balvin", id:"1vyhD5VmyZ7KMfW5gqLgo5"}, {name: "Bad Bunny", id:"4q3ewBCX7sLwd24euuV69X"}, {name: "Maluma", id:"1r4hJ1h58CWwUQe3MxPuau"}, {name: "Rosal√≠a", id:"7ltDVBr6mKbRvohxheJ9h1"}];
// const INITIAL_STATE = {songs:[], artists: SEED_ARTISTS, albums:[]};
import {GET_TRANSLATION} from "./actionTypes";
const ERRORS = {searchError: false, lyricsError: false, translationError: false}
const INITIAL_STATE = {translation: "", errors: ERRORS }

function rootReducer(state=INITIAL_STATE, action) {
    switch(action.type) {
        case GET_TRANSLATION:
        
            // const updatedStateWithErrors = {...state, state.errors.lyricsError = action.errors.lyricsError}
        return {...state, translation: action.translation}

        default:
            return state;
    }
}

export default rootReducer;
