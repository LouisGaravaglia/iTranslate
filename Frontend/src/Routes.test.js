import React from 'react'
import {render, fireEvent, waitFor, screen, cleanup} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import { createStore } from "redux";
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import rootReducer from "./reducers/rootReducer";
import Routes from "./Routes";

//NEED TO IMPLEMENT A MOC INTERSECTION OBSERVER IN ORDER FOR TESTS TO PASS
const mockIntersectionObserver = class {
  constructor() {}
  observe() {}
  unobserve() {}
  disconnect() {}
};

beforeEach(() => {
  window.IntersectionObserver = mockIntersectionObserver;
});

afterEach(cleanup);

const startingState = {
  selectedTrack: {trackName: "chanel", trackId: 42}, 
  errors: {translationError: false, languageError: false, lyricsError: false, searchError: false, generalError: false}
  };

//REDUCER FUNCTION TO MIMIC REDUX REDUCER FOR STORE
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_TRACKS":
      return {...state, tracks: action.tracks};
    default:
      return state;
  };
};

//FUNCTION TO ALLOW THE COMPONENT TO BE RENDERED USING OUR MAKESHIF REDUX STORE
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for Routes component', () => {

  it('renders without crashing', () => {
    const history = createMemoryHistory();

    renderWithRedux(
      <Router history={history}>
        <Routes />
      </Router>
    );
  });
});

//SNAPSHOT TEST
describe('Snapshot Test for Routes component', () => {

  it('matches snapshot', () => {
    const history = createMemoryHistory();

    const {asFragment} = renderWithRedux(
      <Router history={history}>
        <Routes />
      </Router>)
    expect(asFragment()).toMatchSnapshot();
  });

});
