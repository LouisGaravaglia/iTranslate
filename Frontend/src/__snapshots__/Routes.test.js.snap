// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Routes component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="Flash-Messages-Container"
    />
    <div
      style="background-color: rgb(29, 77, 190);"
    >
      SearchLanding handleTrackSearchSubmit=/&gt;
    </div>
  </div>
</DocumentFragment>
`;
